src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+                book.setId(id);
+            }
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book wasn't added. 0 rows were affected");
+            }
+            statement.close();
+            connection.close();
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error while adding book", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Connection connection = ConnectionUtil.getConnection();
+        String sqlSelect = "SELECT * FROM books WHERE id = ?";
+        try {
+            PreparedStatement statement = connection.prepareStatement(sqlSelect);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (!resultSet.first()) {
+                return Optional.empty();
+            }
+            statement.close();
+            connection.close();
+            return Optional.of(getBookFromResultSet(resultSet));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred wile finding by id", e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        Connection connection = ConnectionUtil.getConnection();
+        String sqlSelect = "SELECT * FROM books";
+        try {
+            PreparedStatement statement = connection.prepareStatement(sqlSelect);
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> res = new ArrayList<>();
+            while (resultSet.next()) {
+                res.add(getBookFromResultSet(resultSet));
+            }
+            statement.close();
+            connection.close();
+            return res;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred while getting all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        Connection connection = ConnectionUtil.getConnection();
+        String sqlUpdate = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(sqlUpdate);
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            int affectedRows = preparedStatement.executeUpdate();
+            preparedStatement.close();
+            connection.close();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Book wasn't changed. 0 rows were affected");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error occurred while updating book", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        Connection connection = ConnectionUtil.getConnection();
```

Comment:
you should use `try-with-resources` for both `Connection` and `PreparedStatement`